Add usedevelop flag for coverage-tox issue

dorukozturk opened the pull request:
## Motivation

Fixes #269 

Coverage was reporting 0% even if the coverage was ran. Running tox environment in development mode and correctly referencing pynwb path fixed this issue. 

## Checklist

- [x] Have you checked our [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR description clearly describes problem and the solution?
- [x] Is your contribution compliant with our coding style ? This can be checked running `flake8` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/NeurodataWithoutBorders/pynwb/pulls) for the same change?
- [x] Have you included the relevant issue number using `#XXX` notation where `XXX` is the issue number ?

Commit:
Add usedevelop flag for coverage-tox issue

jcfr commented:
Hi,

Thanks. Look like the _version.py script wasn't we excluded from the
report. Is the branch on the latest dev ?

May the config setting is ignored ?

(Can't easily check, on my phone)

Jc

On Nov 15, 2017 3:59 PM, "Doruk Ozturk" <notifications@github.com> wrote:

@dorukozturk <https://github.com/dorukozturk> requested your review on:
NeurodataWithoutBorders/pynwb#271
<https://github.com/NeurodataWithoutBorders/pynwb/pull/271> Add usedevelop
flag for coverage-tox issue.

—
You are receiving this because your review was requested.

Reply to this email directly, view it on GitHub
<https://github.com/NeurodataWithoutBorders/pynwb/pull/271#event-1343865534>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AANXo4K8R5SRV20kUfbu2Tww-bPpDJOjks5s21DDgaJpZM4QfXeG>
.

jcfr commented:
Not sure why is has to be done in developer mode.

Here is what I used to do to make sure all codecov variables are passed: https://github.com/scikit-build/cmake-python-distributions/blob/a6ed3f1c89a88c3968792d167313716dbb96b33b/.circleci-matrix.yml#L17

May be related is #272

jcfr commented on .circleci/config.yml:
```diff
@@ -63,21 +63,24 @@ jobs:
     environment:
-     - TOX_ENV: "py35,CI-py35"
+     - TEST_TOX_ENV: "py35,CI-py35"
+     - COVERAGE_TOX_ENV: "py35"
```
Are the variable switched ? Look like `COVERAGE_TOX_ENV` should be associated with `CI-py35`

jcfr replied:
Also not clear why it is `py35,CI-py35` instead of `CI-py35`

dorukozturk replied:
I think you are right. I think all envs inherit from testenv so CI-py35 should be enough.

dorukozturk replied:
I think I will refactor tox.ini. We should not need comments there.

Commit:
Fix issue with running coverage twice in circle

Commit:
Fix appveyor coverage problems and integrate tox for windows

dorukozturk commented:
@jcfr I think this is ready. I fixed all the coverage related issues and made appveyor also run tox for consistency. Also fixed the running coverage twice issue and added some comments too.

jcfr commented on appveyor.yml:
```diff
@@ -9,21 +9,44 @@ environment:
       ENV: miniconda
+      TEST_TOX_ENV: py27
+      COVERAGE_TOX_ENV: coverage-py27
+      BUILD_TOX_ENV: build-py27
```
How do you ensure that tox will run into the miniconda env ? 

References:
* https://stackoverflow.com/questions/30555943/is-it-possible-to-use-tox-with-conda-based-python-installations
* https://github.com/tox-dev/tox/tree/master/doc/drafts and also `https://github.com/tox-dev/tox/issues/273`

jcfr commented:
Nice, much cleaner. Also nice to rename the tox env.

The issue with this approach is that tox is called  multiple times (once for build, once for test wheel install, once for coverage)

On windows, this kills the CI speed (which was not already great). Now at 20 minutes

I still think have one tox env with conditional execution of command would be the way to go.

jcfr commented:
> The issue with this approach is that tox is called multiple times (once for build, once for test wheel install, 
> once for coverage)

> On windows, this kills the CI speed (which was not already great). Now at 20 minutes

> I still think have one tox env with conditional execution of command would be the way to go.

Having an other look. I suggest we move forward with this and improve in follow up PR.

We may have to revisit the miniconda testing

dorukozturk commented:
@jcfr I agree. I can work on conda-tox stuff on another PR.
